pkg/controller/atlasproject/integrations.go

```diff
@@ -148,41 +147,21 @@ func (r *AtlasProjectReconciler) checkIntegrationsReady(ctx *workflow.Context, n
 	return true
 }
 
-func AreIntegrationsEqual(atlas, specAsAtlas *aliasThirdPartyIntegration) bool {
-	return reflect.DeepEqual(cleanCopyToCompare(atlas), cleanCopyToCompare(specAsAtlas))
-}
-
-func cleanCopyToCompare(input *aliasThirdPartyIntegration) *aliasThirdPartyIntegration {
-	if input == nil {
-		return input
-	}
-
-	result := *input
-	keepLastFourChars(&result.APIKey)
-	keepLastFourChars(&result.APIToken)
-	keepLastFourChars(&result.LicenseKey)
-	keepLastFourChars(&result.Password)
-	keepLastFourChars(&result.ReadToken)
-	keepLastFourChars(&result.RoutingKey)
-	keepLastFourChars(&result.Secret)
-	keepLastFourChars(&result.ServiceKey)
-	keepLastFourChars(&result.WriteToken)
-
-	return &result
+func integrationsApplied(_, _ *aliasThirdPartyIntegration) bool {
+	// As integration secrets are redacted from Alas, we cannot properly compare them,
+	// so as a simple fix here we assume changes were applied correctly as we would
+	// have otherwise errored out as are always needed
+	// TODO: remove and replace calls to this with areIntegrationsEqual when
+	//       that code is properly comparing fields
+	return true
 }
 
-func keepLastFourChars(strPtr *string) {
-	if strPtr == nil {
-		return
-	}
-
-	charCount := 4
-	str := *strPtr
-	if len(str) <= charCount {
-		return
-	}
-
-	*strPtr = str[len(str)-charCount:]
+func areIntegrationsEqual(_, _ *aliasThirdPartyIntegration) bool {
```

Comment:
It's probably a matter of taste but I would extract this into an interface and provide an "AlwaysApplyComparer" or something along these lines. Just to get this stop-the-bleeding patch code out of the sight. 

Feel free to ignore this suggestion if you think otherwise.

Reply:
Not sure is worth it. We are going to actually write quite some new code once we need to actually compare. We just do now know what that code is now, so having interfaces seems premature. For instance, if we go for implicit evaluation these interfaces break, because we need input from 3 sources, Atlas, Spec and annotations.